invoices/invoiceregistry.go

```diff
@@ -692,8 +717,28 @@ func (i *InvoiceRegistry) processKeySend(ctx invoiceUpdateCtx) error {
 	_, err = i.AddInvoice(invoice, ctx.hash)
 	if err != nil && err != channeldb.ErrDuplicateInvoice {
 		return err
+	} else if err == channeldb.ErrDuplicateInvoice {
```

Comment:
Refactor with a `switch` that first checks for `ErrDuplicateInvoice`?

Reply:
done